toppingmaker/projecttopping.py

```diff
@@ -379,11 +382,80 @@ def make_items(
 
                 self[name] = maptheme_item
 
+    class Variables(dict):
+        """
+        A dict object of dict items describing a variable according to the variable keys listed in the ExportSettings passed on parsing the QGIS project.
+        """
+
+        def make_items(
+            self,
+            project: QgsProject,
+            export_settings: ExportSettings,
+        ):
+            self.clear()
+            for variable_key in export_settings.variables:
+                self[variable_key] = QgsExpressionContextUtils.projectScope(
+                    project
+                ).variable(variable_key)
+
+    class Layouts(dict):
+        """
+        A dict object of dict items describing a layout with templatefile according to the layout names listed in the ExportSettings passed on parsing the QGIS project.
+        """
+
+        class LayoutItemProperties(object):
```

Comment:
Do we want to have nested classes here?
For simplicity I'd keep classes unnested (unless it's really an internal implementation detail?)

Reply:
My thought was that in future we might have other properties than just the template file, like e.g. layout items etc. (manually defined) but I do agree that at the moment it's an overkill.